apis/ec2/src/main/java/org/jclouds/ec2/compute/EC2ComputeService.java

```diff
@@ -220,10 +223,16 @@ void deleteSecurityGroup(String region, String group) {
       checkNotNull(emptyToNull(group), "group must be defined");
       String groupName = namingConvention.create().sharedNameForGroup(group);
 
-      if (!client.getSecurityGroupApi().get().describeSecurityGroupsInRegion(region, groupName).isEmpty()) {
+      Multimap<String, String> securityGroupFilterByName = ImmutableMultimap.of("group-name", groupName);
+      Set<SecurityGroup> securityGroupsToDelete = client.getSecurityGroupApi().get()
+              .describeSecurityGroupsInRegionWithFilter(region, securityGroupFilterByName);
```

Comment:
@neykov will your comment about security group name filtering [here](https://github.com/jclouds/jclouds/pull/1091/files#r113952626) also apply in this code?